logstash-core/lib/logstash/api/commands/stats.rb

```diff
@@ -92,9 +108,13 @@ def hot_threads(options={})
         end
 
         private
-        def plugins_stats_report(pipeline_id)
+        def plugins_stats_report(pipeline_id, opts)
           stats = service.get_shallow(:stats, :pipelines, pipeline_id.to_sym)
-          PluginsStats.report(stats)
+          extended_stats = LogStash::Config::PipelinesInfo.format_pipelines_info(
+            service.agent,
+            service.snapshot.metric_store,
+            true)
+          PluginsStats.report(stats, extended_stats.shift, opts)
```

Comment:
It is unclear to me what we are attempting to accomplish with this `Array#shift` sent to `extended_stats`.

From my reading, `LogStash::Config::PipelinesInfo::format_pipelines_info` is returning an array of data for _all_ of the running pipelines, so returning the first element is not guaranteed to pick off the extended stats that match the `pipeline_id` that has been passed to this method.

Reply:
Gah! Good catch. I had put this in for testing and forgot to remove it. I changed it so that we should now pass in the correct pipeline.